packages/flutter_tools/lib/src/context_runner.dart

```diff
@@ -47,6 +47,7 @@ Future<T> runInContext<T>(
       AndroidSdk: AndroidSdk.locateAndroidSdk,
       AndroidStudio: AndroidStudio.latestValid,
       AndroidWorkflow: () => new AndroidWorkflow(),
+      AndroidValidator: () => new AndroidValidator(),
```

Comment:
What's the difference between a validator and a workflow?

Reply:
Currently, the workflow just checks whether we should run the validators associated with it.

I thought it'd be strange to split a validator and have only one of them with the workflow attached. If it'd be better to keep the workflow on the validator, I don't mind undoing this part of the change.

Reply:
I didn't mean for you to undo this change but it would be good to have a better explanation of Validator vs Workflow in comments. 

So a workflow is just a collection of validators then?

This is confusing because down below, you have a list of workflows and a list of validators in the same class. Are these just validators that don't belong to any workflow?

Reply:
I'm more or less breaking the old, implicit relationship where a workflow is-a validator (both workflow classes were also validators and DoctorValidatorProvider also made that assumption), so they're now two totally separate classes. A validator is still a validator (checks for installed software and generates doctor output), while a workflow checks if the environment is one where we can list or launch devices. 

The only relation between validators and workflows now is that we use Workflow.appliesToHostPlatform to decide whether to add a validator to the validator list (skipping the iOS validator if we're not on Mac).

The validators in the provider are effectively the same as they used to be. The new workflows list is there because I can no longer have the validators list do double duty, so I'm explicitly listing them instead of using the old code that got them implicitly by checking which validators were also workflows.

Workflow might be a bad name for it now, since the class scope is a lot smaller now that the validator methods have been pulled out. But I don't know what would be a good, unconfusing name.

Reply:
Got it. Thanks for the context.